packages/server/src/sdk/app/tables/migration.ts

```diff
@@ -24,25 +25,58 @@ export interface MigrationResult {
 
 export async function migrate(
   table: Table,
-  oldColumn: FieldSchema,
-  newColumn: FieldSchema
+  oldColumnName: string,
+  newColumnName: string
 ): Promise<MigrationResult> {
-  if (newColumn.name in table.schema) {
-    throw new BadRequestError(`Column "${newColumn.name}" already exists`)
+  if (newColumnName in table.schema) {
+    throw new BadRequestError(`Column "${newColumnName}" already exists`)
   }
 
-  if (newColumn.name === "") {
+  if (newColumnName === "") {
     throw new BadRequestError(`Column name cannot be empty`)
   }
 
-  if (dbCore.isInternalColumnName(newColumn.name)) {
+  if (dbCore.isInternalColumnName(newColumnName)) {
     throw new BadRequestError(`Column name cannot be a reserved column name`)
   }
 
+  const oldColumn = table.schema[oldColumnName]
+
+  if (!oldColumn) {
+    throw new BadRequestError(
+      `Column "${oldColumnName}" does not exist on table "${table.name}"`
+    )
+  }
+
+  if (
+    oldColumn.type !== FieldType.LINK ||
+    oldColumn.tableId !== InternalTable.USER_METADATA
+  ) {
+    throw new BadRequestError(
+      `Only user relationship migration columns is currently supported`
+    )
+  }
+
+  const type =
+    oldColumn.relationshipType === RelationshipType.ONE_TO_MANY
+      ? FieldType.BB_REFERENCE_SINGLE
+      : FieldType.BB_REFERENCE
+  const newColumn: FieldSchema = {
+    name: newColumnName,
+    type,
+    subtype: BBReferenceFieldSubType.USER,
+  }
```

Comment:
When I made this API originally the idea was to get the caller to specify the full FieldSchema for the new column, because it kept the API open to doing other types of migration in the future.

I'm not sure I follow why exactly it's being changed away from that. Won't this change make it difficult to extend this in future?

Reply:
If we follow this way, we need the frontend to know about the backend inside-outs. I made these changes in order to keep it as a backend responsibility (depending on the new column type). This also makes it easier to test, as otherwise we would be accepting anything from the frontend

Reply:
That's fair, I did kinda wonder if I was making it more complicated than it needed to be in order to fit use-cases that didn't exist yet.